### ISSUE TITLE
Top-level `oneOf` makes fastify crash during startup

### ISSUE THREAD

**Issue by fox1t** [fastify/fast-json-stringify#262]
## 🐛 Bug Report

Top-level `oneOf` throws `undefined unsupported`.

## To Reproduce

This response schema throws the error:
```js
const schema = {
  oneOf: [
    {
      type: 'object',
      additionalProperties: false,
      properties: {
        code: {
          type: 'string',
          pattern: '^[0-9A-Za-z_]+$',
        },
      },
    },
    {
      type: 'object',
      additionalProperties: false,
      properties: {
        name: {
          type: 'object',
          additionalProperties: false,
          patternProperties: {
            '^[a-z]{2}_[A-Z]{2}$': {
              type: 'string',
            },
          },
          properties: {
            default: {
              type: 'string',
            },
          },
        },
      },
    },
  ],
}
```

## Expected behavior

No crash at fastify startup.

## Your Environment

- *node version*: 14
- *fastify version*: >=3.0.0
- *os*: macOS

**Comment by mcollina** [comment]
Nice!

**Comment by fox1t** [comment]
Any hints on how this could be fixed/implemented? I tried to look into this issue and I found that the code for handling allOf/anyOf/oneOf is only built when inside `properties` in `type === 'object'`.

What do you think could be the best approach?

**Comment by mcollina** [comment]
This needs a biggish refactoring moving out that logic from `type === 'object'`

**Comment by Adibla** [comment]
It seems to be fixed in [1c6b4da08f84a8daf7579595ec9263aa75d92c52](https://github.com/Adibla/fast-json-stringify/commit/1c6b4da08f84a8daf7579595ec9263aa75d92c52)

**Comment by mcollina** [comment]
thanks!
